apps/src/applab/applab.js

```diff
@@ -814,6 +814,12 @@ function setupReduxSubscribers(store) {
       );
     }
 
+    const lastIsPreview = lastState.data && lastState.data.isPreviewOpen;
+    const isPreview = state.data && state.data.isPreviewOpen;
+    if (isDataMode && isPreview && !lastIsPreview) {
```

Comment:
What use case does including the !lastIsPreview prevent?

Reply:
This function is called when any action occurs. We don't want to re-fetch data unnecessarily if the preview tab is already open, so this conditional captures the state change from closed -> open and fetches data only in that case.

Reply:
I'm concerned that this might be extending this function to the point of making it fragile. Is it worth trying to break out some of the functionality here into helpers or state independent functions?

Reply:
This is the same pattern as line 807 and line 825, so I don't think this makes anything worse.

We have https://codedotorg.atlassian.net/browse/STAR-675 to track whether we can clean this up with thunks, but I don't think we should hold this change on that ticket.

Reply:
Ahh got it - good to know it's tracked.